packages/compiler/src/core/types.ts

```diff
@@ -285,6 +291,14 @@ export interface ModelProperty extends BaseType, DecoratedType {
   model?: Model;
 }
 
+export interface ModelValidate extends BaseType, DecoratedType {
+  kind: "ModelValidate";
+  node: ModelValidateNode;
```

Comment:
I guess with this implementation we expect emitters to grab the node and party on from there. I wonder if we should instead define a separate AST structure for validates expressions? Part of this is because we don't consider the AST to be "public API" and therefore heavily discourage usage of this structure in emitters. But that aside, this could have a few possible benefits:

1: potentially simpler structure (e.g. probably don't need the top-level node but its ProjectionExpression value, don't need node flags, don't need directives, etc.)
2: offer text ranges relative to just the expression as opposed to the entire TypeSpec source file
3: include useful things from the check phase e.g. method invocation can have an actual target value you can use for comparison so you don't need to chug on a member expression yourself